File: packages/typescript/src/endpoints/plugin_marketplace.ts
/**
 * Plugin_Marketplace API Client
 */

import { AxiosInstance } from 'axios'

export class Plugin_MarketplaceClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * GET /marketplace
   */
  async getMarketplace(plugin_type?: string, category?: string, pricing_tier?: string, search?: string, is_official?: boolean, is_vetted?: boolean, min_rating?: number, page?: number, per_page?: number, sort_by?: string, sort_order?: string): Promise<PluginListResponse> {
        const params = {'plugin_type: plugin_type, category: category, pricing_tier: pricing_tier, search: search, is_official: is_official, is_vetted: is_vetted, min_rating: min_rating, page: page, per_page: per_page, sort_by: sort_by, sort_order: sort_order'}
        const data = undefined
        return this.httpClient.get(`/marketplace`, params)
  }
  /**
   * GET /{plugin_id}
   */
  async get{plugin_id}(plugin_id: string): Promise<PluginResponse> {
        const params = {'plugin_id: plugin_id'}
        const data = undefined
        return this.httpClient.get(`/{plugin_id}`, params)
  }
  /**
   * POST /{plugin_id}/install
   */
  async post{plugin_id}install(plugin_id: string, organization_id?: string): Promise<PluginInstallationResponse> {
        const params = {}
        const data = {'plugin_id: plugin_id, organization_id: organization_id'}
        return this.httpClient.post(`/{plugin_id}/install`, data, params)
  }
  /**
   * GET /{plugin_id}/download
   */
  async get{plugin_id}download(plugin_id: string, version?: string): Promise<any> {
        const params = {'plugin_id: plugin_id, version: version'}
        const data = undefined
        return this.httpClient.get(`/{plugin_id}/download`, params)
  }
  /**
   * GET /installed
   */
  async getInstalled(): Promise<List[PluginInstallationResponse]> {
        const params = {}
        const data = undefined
        return this.httpClient.get(`/installed`, params)
  }
  /**
   * PUT /{plugin_id}
   */
  async put{plugin_id}(plugin_id: string, plugin_data: PluginUpdate): Promise<PluginResponse> {
        const params = {}
        const data = {'plugin_id: plugin_id, plugin_data: plugin_data'}
        return this.httpClient.put(`/{plugin_id}`, data, params)
  }
  /**
   * DELETE /{plugin_id}
   */
  async delete{plugin_id}(plugin_id: string): Promise<any> {
        const params = {}
        const data = plugin_id
        return this.httpClient.delete(`/{plugin_id}`, data, params)
  }
  /**
   * POST /{plugin_id}/publish
   */
  async post{plugin_id}publish(plugin_id: string): Promise<PluginResponse> {
        const params = {}
        const data = plugin_id
        return this.httpClient.post(`/{plugin_id}/publish`, data, params)
  }
  /**
   * POST /{plugin_id}/versions
   */
  async post{plugin_id}versions(plugin_id: string, version_data: PluginVersionCreate): Promise<PluginVersionResponse> {
        const params = {}
        const data = {'plugin_id: plugin_id, version_data: version_data'}
        return this.httpClient.post(`/{plugin_id}/versions`, data, params)
  }
  /**
   * GET /{plugin_id}/versions
   */
  async get{plugin_id}versions(plugin_id: string): Promise<List[PluginVersionResponse]> {
        const params = {'plugin_id: plugin_id'}
        const data = undefined
        return this.httpClient.get(`/{plugin_id}/versions`, params)
  }
  /**
   * GET /{plugin_id}/reviews
   */
  async get{plugin_id}reviews(plugin_id: string, page?: number, limit?: number, min_rating?: number, sort_by?: string, sort_order?: string): Promise<List[PluginReviewResponse]> {
        const params = {'plugin_id: plugin_id, page: page, limit: limit, min_rating: min_rating, sort_by: sort_by, sort_order: sort_order'}
        const data = undefined
        return this.httpClient.get(`/{plugin_id}/reviews`, params)
  }
  /**
   * POST /{plugin_id}/reviews
   */
  async post{plugin_id}reviews(plugin_id: string, review_data: PluginReviewCreate): Promise<PluginReviewResponse> {
        const params = {}
        const data = {'plugin_id: plugin_id, review_data: review_data'}
        return this.httpClient.post(`/{plugin_id}/reviews`, data, params)
  }
  /**
   * GET /{plugin_id}/reviews/{review_id}
   */
  async get{plugin_id}reviews{review_id}(plugin_id: string, review_id: string): Promise<PluginReviewResponse> {
        const params = {'plugin_id: plugin_id, review_id: review_id'}
        const data = undefined
        return this.httpClient.get(`/{plugin_id}/reviews/{review_id}`, params)
  }
  /**
   * PUT /{plugin_id}/reviews/{review_id}
   */
  async put{plugin_id}reviews{review_id}(plugin_id: string, review_id: string, review_update: PluginReviewUpdate): Promise<PluginReviewResponse> {
        const params = {}
        const data = {'plugin_id: plugin_id, review_id: review_id, review_update: review_update'}
        return this.httpClient.put(`/{plugin_id}/reviews/{review_id}`, data, params)
  }
  /**
   * DELETE /{plugin_id}/reviews/{review_id}
   */
  async delete{plugin_id}reviews{review_id}(plugin_id: string, review_id: string): Promise<any> {
        const params = {}
        const data = {'plugin_id: plugin_id, review_id: review_id'}
        return this.httpClient.delete(`/{plugin_id}/reviews/{review_id}`, data, params)
  }
  /**
   * POST /{plugin_id}/reviews/{review_id}/helpful
   */
  async post{plugin_id}reviews{review_id}helpful(plugin_id: string, review_id: string): Promise<any> {
        const params = {}
        const data = {'plugin_id: plugin_id, review_id: review_id'}
        return this.httpClient.post(`/{plugin_id}/reviews/{review_id}/helpful`, data, params)
  }
}

File: packages/python/src/endpoints/plugin_marketplace.py
"""
Plugin_Marketplace API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class Plugin_MarketplaceClient:
    """Client for plugin_marketplace endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def get_marketplace(self, plugin_type: Optional[str] = None, category: Optional[str] = None, pricing_tier: Optional[str] = None, search: Optional[str] = None, is_official: Optional[bool] = None, is_vetted: Optional[bool] = None, min_rating: Optional[float] = None, page: Optional[float] = None, per_page: Optional[float] = None, sort_by: Optional[str] = None, sort_order: Optional[str] = None):
        """GET /marketplace"""
        url = f"/marketplace"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_plugin_id(self, plugin_id: str):
        """GET /{plugin_id}"""
        url = f"/{plugin_id}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_plugin_id_install(self, plugin_id: str, organization_id: Optional[str] = None):
        """POST /{plugin_id}/install"""
        url = f"/{plugin_id}/install"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def get_plugin_id_download(self, plugin_id: str, version: Optional[str] = None):
        """GET /{plugin_id}/download"""
        url = f"/{plugin_id}/download"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_installed(self):
        """GET /installed"""
        url = f"/installed"
        params = {}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def put_plugin_id(self, plugin_id: str, plugin_data: Any):
        """PUT /{plugin_id}"""
        url = f"/{plugin_id}"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("PUT", url, params=params, json=data)


    async def delete_plugin_id(self, plugin_id: str):
        """DELETE /{plugin_id}"""
        url = f"/{plugin_id}"
        params = {}
        data = plugin_id
        return await self._request("DELETE", url, params=params, json=data)


    async def post_plugin_id_publish(self, plugin_id: str):
        """POST /{plugin_id}/publish"""
        url = f"/{plugin_id}/publish"
        params = {}
        data = plugin_id
        return await self._request("POST", url, params=params, json=data)


    async def post_plugin_id_versions(self, plugin_id: str, version_data: Any):
        """POST /{plugin_id}/versions"""
        url = f"/{plugin_id}/versions"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def get_plugin_id_versions(self, plugin_id: str):
        """GET /{plugin_id}/versions"""
        url = f"/{plugin_id}/versions"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_plugin_id_reviews(self, plugin_id: str, page: Optional[float] = None, limit: Optional[float] = None, min_rating: Optional[float] = None, sort_by: Optional[str] = None, sort_order: Optional[str] = None):
        """GET /{plugin_id}/reviews"""
        url = f"/{plugin_id}/reviews"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_plugin_id_reviews(self, plugin_id: str, review_data: Any):
        """POST /{plugin_id}/reviews"""
        url = f"/{plugin_id}/reviews"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def get_plugin_id_reviews_review_id(self, plugin_id: str, review_id: str):
        """GET /{plugin_id}/reviews/{review_id}"""
        url = f"/{plugin_id}/reviews/{review_id}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def put_plugin_id_reviews_review_id(self, plugin_id: str, review_id: str, review_update: Any):
        """PUT /{plugin_id}/reviews/{review_id}"""
        url = f"/{plugin_id}/reviews/{review_id}"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("PUT", url, params=params, json=data)


    async def delete_plugin_id_reviews_review_id(self, plugin_id: str, review_id: str):
        """DELETE /{plugin_id}/reviews/{review_id}"""
        url = f"/{plugin_id}/reviews/{review_id}"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("DELETE", url, params=params, json=data)


    async def post_plugin_id_reviews_review_id_helpful(self, plugin_id: str, review_id: str):
        """POST /{plugin_id}/reviews/{review_id}/helpful"""
        url = f"/{plugin_id}/reviews/{review_id}/helpful"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)

